src/main/kotlin/com/yapp/weekand/domain/auth/service/AuthService.kt

```diff
@@ -47,10 +53,39 @@ class AuthService (
 	}
 
 	fun reissueAccessToken(refreshToken: String): ReissueAccessTokenResponse {
-		val email: String = redisService.getValue(refreshToken)
+		val email: String = redisService.getValue("$REFRESH_TOKEN_PREFIX:$refreshToken")
 			?: throw InvalidTokenException()
 
 		userRepository.findByEmail(email) ?: throw UserNotFoundException()
 		return ReissueAccessTokenResponse(jwtProvider.createAccessToken(email))
 	}
+
+	fun sendEmailAuthKey(email: String) {
+		if (userRepository.existsUserByEmail(email)) {
+			throw UserNotFoundException()
+		}
+		val authKey = createAuthKey()
+		redisService.setValue("$AUTH_KEY_PREFIX:$email", authKey, authKeyExpiry)
+		emailService.sendEmail(email, authKey, "인증번호 안내")
+	}
+
+	fun isValidAuthKey(request: ValidAuthKeyInput): Boolean {
+		val authKey = redisService.getValue("$AUTH_KEY_PREFIX:"+request.email)
+		if (authKey != null && request.authKey == authKey) {
+			return true
+		}
+		return false
```

Comment:
뭔가 각 비교문이 AND 조건문 등으로 엮이면 가독성이 떨어지는거 같아요
이렇게 해보면 어떨까요?
```suggestion
        if (authKey == null) {
			return false
		}
        if(authKey != request.authKey) {
            return false
        }
		return true
```

Reply:
이 부분 고민하다가 간결한게 좋겠다 싶어서 저렇게 작성했었는데 가독성 측면에서 제안해주신 방법이 더 좋은 것 같네요..!
아래와 같이 null 확인 부분만 수정해서 반영하겠습니다!

```kotlin 
val authKey = redisService.getValue("$AUTH_KEY_PREFIX:"+request.email) 
		  ?: return false
```